### DISCUSSION TITLE
The status of current pocl auto-vectorization enablement, especially for ARM SVE(Scalable Vector Extension) or RISC-V RVV

### DISCUSSION THREAD

**Discussion by soccercheng** [pocl/pocl#1561] [Q&A]
Hi, POCL Team

I'm curious about the current auto-vectorization enablement of pocl using LLVM to compile/optimize OpenCL C to target CPU architecture, like ARM SVE or even RISC-V RVV. 

- Is there any benchmark number available for reference?  or
- TO-DO List for auto-vectorization enablement plan ?

**Comment by pjaaskel** [comment]
The autovectorizer cleanup/rework is ongoing. The latest PR: https://github.com/pocl/pocl/pull/1559  The plan is still that we will utilize LLVM loopvec as usually, just helping it and remove any hindrances blocking it from doing its job.  Thus you should look for LLVM's RVV loop autovec and see if you can help improving that to make things better also for PoCL autovec.
